File: app/src/main/java/com/vkochenkov/weatherfromopenapis/presentation/recycler/CityClickListener.java
package com.vkochenkov.weatherfromopenapis.presentation.recycler;

import android.view.View;

import com.vkochenkov.weatherfromopenapis.data.db.entities.City;

public interface CityClickListener {
    View.OnClickListener onCityClickListener(City city);

    View.OnLongClickListener onCityLongClickListener(int position, City city);
}

File: app/src/main/java/com/vkochenkov/weatherfromopenapis/presentation/recycler/CityClickListener.kt
package com.vkochenkov.weatherfromopenapis.presentation.recycler

import android.view.View
import android.view.View.OnLongClickListener
import com.vkochenkov.weatherfromopenapis.data.db.entities.City

interface CityClickListener {
    fun onCityClickListener(city: City): View.OnClickListener
    fun onCityLongClickListener(position: Int, city: City): OnLongClickListener
}